File: SNAKPROC.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    int R, L, g = 0;
    cin >> R;
    string s;
    
    if (R <= 500 && R >= 1)
    {
        for (int i = 0; i < R; i++)
        {
            cin >> L;
            if (L <= 500 && L >= 1)
            {
                g = 0;
                cin >> s;
                for (int j = 0; j < L; j++)
                {
                    if (s[j] == 'H')
                        g++;
                    else if (s[j] == 'T')
                        g--;
                    if ( g > 1 || g < 0)
                    {
                        cout << "Invalid\n";
                        break;
                    }
                }
                if(g == 0)
                    cout << "Valid\n";
                else if (g == 1)
                    cout << "Invalid\n";
            }
        }
    }
	return 0;
}

File: SNAKPROC.py
if __name__ == "__main__":
    R = int(input())
    g = 0
    
    if R in range(1, 501):
        for i in range(R):
            g = 0
            L = int(input())
            if L in range(1, 501):
                lst = input()
                for j in range(L):
                    if lst[j] == 'H':
                        g = g + 1
                    elif lst[j] == 'T':
                        g = g - 1
                    if g > 1 or g < 0:
                        print ("Invalid")
                        break
                if g == 0:
                    print ("Valid")
                elif g == 1:
                    print ("Invalid")